magellan-sample-advanced/src/main/java/com/wealthfront/magellan/sample/advanced/MainActivity.kt

```diff
@@ -29,51 +16,9 @@ class MainActivity : AppCompatActivity(), ActionBarConfigListener {
     setContentScreen(expedition)
   }
 
-  override fun onCreateOptionsMenu(menu: Menu): Boolean {
-    menuInflater.inflate(R.menu.menu, menu)
-    expedition.setMenu(menu)
-    return true
-  }
-
-  override fun onPrepareOptionsMenu(menu: Menu): Boolean {
-    expedition.setMenu(menu)
-    return true
-  }
-
   override fun onBackPressed() {
     if (!expedition.backPressed()) {
       super.onBackPressed()
     }
   }
-
-  @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
-  override fun onNavigate(actionBarConfig: ActionBarConfig?) {
-    if (actionBarConfig != null) {
-      updateStatusBar(actionBarConfig.colorRes())
-      actionBar?.show()
-    } else {
-      actionBar?.hide()
-    }
-  }
-
-  @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
-  private fun updateStatusBar(@ColorRes actionBarColorRes: Int = DEFAULT_ACTION_BAR_COLOR_RES) {
-    if (actionBarColorRes == DEFAULT_ACTION_BAR_COLOR_RES) {
-      animateStatusBarColor(window, window.statusBarColor, R.color.colorAccent)
-    } else {
-      val statusBarColor = ContextCompat.getColor(this, actionBarColorRes)
-      animateStatusBarColor(window, window.statusBarColor, statusBarColor)
-    }
-  }
-
-  @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
-  private fun animateStatusBarColor(window: Window, startColor: Int, endColor: Int) {
-    if (startColor == endColor) {
-      return
-    }
-    val statusColorAnimator = ValueAnimator.ofObject(ArgbEvaluator(), startColor, endColor)
-    statusColorAnimator.duration = DEFAULT_ANIM_DURATION_MS
-    statusColorAnimator.addUpdateListener { window.statusBarColor = it.animatedValue as Int }
-    statusColorAnimator.start()
-  }
```

Comment:
Yesssss move all of this out of the `MainActivity` and into a proper Step! This will make things much easier

Reply:
Yup!